### ISSUE TITLE
Colima/Lima broken and unusable on Intel/amd64

### ISSUE THREAD

**Issue by damienmckenna** [ddev/ddev#5298]
**TL;DR: Colima/Lima are broken in latest versions. There are workarounds:**
**See**
* https://github.com/abiosoft/colima/issues/786#issuecomment-1693629650

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
Running bash [-c /var/folders/0m/tg6xfqy14mn4lnws2jmstr4w0000gr/T/test_ddev.sh] 
======= Existing project config =========
These config files were loaded for project myproject-d8: [/Users/dmckenna/Sites/myproject_d8/.ddev/config.yaml]
name: myproject-d8
type: drupal9
docroot: web
php_version: 8.0
webserver_type: nginx-fpm
webimage: ddev/ddev-webserver:20230803_php_serialize_precision
router_http_port: 80
router_https_port: 443
additional_hostnames: []
additional_fqdns: [local.pe.gatech.edu]
mariadb_version: 10.3
database: {mariadb 10.3}
hooks: map[post-import-db:[map[exec:drush cr] map[exec:drush en -y stage_file_proxy] map[exec:drush sql-sanitize -y] map[exec:drush uli]] post-pull:[map[exec:drush cr] map[exec:drush en -y stage_file_proxy] map[exec:drush sql-sanitize -y] map[exec:drush uli]] post-start:[map[exec:mysql -uroot -proot -e "CREATE DATABASE IF NOT EXISTS d7; GRANT ALL ON d7.* to 'db'@'%';" service:db]]]
mailhog_port: 8025
mailhog_https_port: 8026
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2
nodejs_version: 18
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-18437 in ../tryddevproject-18437 =========
OS Information: Darwin DMcKennaWork.local 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 arm64
ProductName:		macOS
ProductVersion:		13.5.1
BuildVersion:		22G90
User information: uid=504(dmckenna) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),79(_appserverusr),80(admin),81(_appserveradm),98(_lpadmin),703(com.apple.sharepoint.group.3),702(com.apple.sharepoint.group.2),33(_appstore),100(_lpoperator),204(_developer),250(_analyticsusers),395(com.apple.access_ftp),398(com.apple.access_screensharing),399(com.apple.access_ssh),400(com.apple.access_remote_ae),701(com.apple.sharepoint.group.1)
DDEV version:  ITEM             VALUE                                                
 DDEV version     v1.22.1                                              
 architecture     arm64                                                
 db               ddev/ddev-dbserver-mariadb-10.4:v1.22.0              
 ddev-ssh-agent   ddev/ddev-ssh-agent:v1.22.0                          
 docker           23.0.6                                               
 docker-compose   v2.20.0                                              
 docker-platform  colima                                               
 mutagen          0.17.2                                               
 os               darwin                                               
 router           ddev/ddev-router:v1.22.0                             
 web              ddev/ddev-webserver:20230803_php_serialize_precision 
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=false
omit-containers=[]
performance-mode=mutagen
router-bind-all-interfaces=false
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
auto-restart-containers=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=v2.20.0
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
router=nginx-proxy
wsl2-no-windows-hosts-mgt=false
router-http-port=80
router-https-port=443
traefik-monitor-port=10999

======= DOCKER info =========
docker location: lrwxr-xr-x  1 dmckenna  admin  34 Jul 31 08:16 /opt/homebrew/bin/docker -> ../Cellar/docker/24.0.5/bin/docker
Docker Desktop Version: Print: Entry, ":CFBundleShortVersionString", Does Not Exist
Print: Entry, ":CFBundleVersion", Does Not Exist
Docker Desktop for Mac File build Doesn'tDocker Desktop for Mac Exist, build WillDocker Desktop for Mac Create: build /Applications/Docker.app/Contents/Info.plistDocker Desktop for Mac File build Doesn'tDocker Desktop for Mac Exist, build WillDocker Desktop for Mac Create: build /Applications/Docker.app/Contents/Info.plist
docker version: 
Client: Docker Engine - Community
 Version:           24.0.5
 API version:       1.42 (downgraded from 1.43)
 Go version:        go1.20.6
 Git commit:        ced0996600
 Built:             Wed Jul 19 19:44:22 2023
 OS/Arch:           darwin/arm64
 Context:           default

Server:
 Engine:
  Version:          23.0.6
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       9dbdbd4b6d7681bd18c897a6ba0376073c2a72ff
  Built:            Fri May 12 13:54:36 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.7.0
  GitCommit:        1fbd70374134b891f97ce19c70b6e50c7b9f4e0d
 runc:
  Version:          1.1.7
  GitCommit:        860f061b76bb4fc671f0f9e900f7d80ff93d4eb7
 docker-init:
  Version:          0.19.0
  GitCommit:        
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
Mutagen is installed in ddev, version=0.17.2
--------------------------------------------------------------------------------
Name: myproject-d8
Identifier: sync_GO1YBz3NOXBsy3Q4voYhyLEbwzYYbVc57j4lhwWyUW7
Labels:
	com.ddev.config-hash: da39a3ee5e6b4b0d3255bfef95601890afd80709
	com.ddev.volume-signature: Users-dmckenna-colima-default-1692971901
Configuration:
	Synchronization mode: Default (Two Way Safe)
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: Default (Portable)
	Ignore VCS mode: Default (Propagate)
	Ignores: None
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/dmckenna/Sites/myproject_d8
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Default (Mutagen Data Directory)
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: Yes
	Synchronizable contents:
		26303 directories
		148846 files (10 GB)
		182 symbolic links
Beta:
	URL: docker://ddev-myproject-d8-web/var/www/html
		DOCKER_HOST=unix:///Users/dmckenna/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Default (Mutagen Data Directory)
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: Yes
	Synchronizable contents:
		4 directories
		0 files (0 B)
		0 symbolic links
	Scan problems:
		.git: scan crossed filesystem boundary
		web/sites/default/files: scan crossed filesystem boundary
Status: Applying changes
--------------------------------------------------------------------------------
======= Docker Info =========
Docker platform: colima 
Colima version: 0.5.5 
Using docker context: default (unix:///Users/dmckenna/.colima/default/docker.sock) 
docker-compose: v2.20.0 
Using DOCKER_HOST=unix:///Users/dmckenna/.colima/default/docker.sock 
Docker version: 23.0.6 
Able to run simple container that mounts a volume. 
Able to use internet inside container. 
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                 195.8G     20.2G    165.6G  11% /

PhpMyAdmin (`dba`) is no longer part of DDEV core, please edit your `omit_containers` configuration to remove it
 Container ddev-myproject-d8-redis  Stopped
 Container ddev-myproject-d8-redis  Removed
 Container ddev-myproject-d8-db  Stopped
 Container ddev-myproject-d8-db  Removed
 Container ddev-myproject-d8-web  Stopped
 Container ddev-myproject-d8-web  Removed
 Container ddev-myproject-d8-solr  Stopped
 Container ddev-myproject-d8-solr  Removed
 Network ddev-myproject-d8_default  Removed
Project myproject-d8 has been stopped.
Stopped mutagen daemon
The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again.
Existing docker containers: 
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
Network ddev_default created 
Starting tryddevproject-18437... 
 Container ddev-ssh-agent  Created 
 Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
 Network ddev-tryddevproject-18437_default  Created 
 Container ddev-tryddevproject-18437-web  Created 
 Container ddev-tryddevproject-18437-db  Created 
 Container ddev-tryddevproject-18437-web  Started 
 Container ddev-tryddevproject-18437-db  Started 
You have Mutagen enabled and your 'php' project type doesn't have `upload_dirs` set. 
For faster startup and less disk usage, set upload_dirs to where your user-generated files are stored. 
If this is intended you can disable this warning with `ddev config --disable-upload-dirs-warning`. 
Starting mutagen sync process... This can take some time. 
Mutagen sync flush completed in 1s.
For details on sync status 'ddev mutagen st tryddevproject-18437 -l' 
 Container ddev-router  Created 
 Container ddev-router  Started 
Successfully started tryddevproject-18437 
Project can be reached at https://tryddevproject-18437.ddev.site https://127.0.0.1:32787 
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Fri, 25 Aug 2023 14:10:38 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-18437.ddev.site from outside:
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Fri, 25 Aug 2023 14:10:38 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== full curl of http://tryddevproject-18437.ddev.site from outside:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-18437'
======== Project ownership on host:
drwxr-xr-x  4 dmckenna  staff  128 Aug 25 10:09 ../tryddevproject-18437
======== Project ownership in container:
drwxr-xr-x 4 dmckenna dialout 4096 Aug 25 14:10 /var/www/html
======== In-container filesystem:
Filesystem                     Type 1K-blocks     Used Available Use% Mounted on
/dev/disk/by-label/data-volume ext4 205313000 21357928 173452980  11% /var/www
======== curl again of tryddevproject-18437 from host:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-18437'
Thanks for running the diagnostic. It was successful.
Please provide the output of this script in a new gist at gist.github.com
Running ddev launch in 5 seconds

Please run cleanup after debugging with 'ddev debug testcleanup'
```
</details>         

### Expected Behavior

Running ddev on Ventura with Colima and Mutagen should work correctly.

### Actual Behavior

Starting a (large-ish) project when Mutagen is enabled results in it failing to start.

### Steps To Reproduce

macOS Ventura 13.5.1.
MacBook Pro M1/Max.
Colima installed via Homebrew.
ddev configured to use Mutagen (default).
All software running the latest.
Colima configured to use vz and all vz-related options.
Start a project with a large codebase and Drupal sites/default/files directory.

**Comment by damienmckenna** [comment]
I switched it over to nfs and it's still failing. "ddev debug nfsmount" shows nfs is configured correctly, but starting the project fails:

```
ddev start
Starting myproject... 
User-managed /Users/dmckenna/Sites/myproject/.ddev/.gitignore will not be managed/overwritten by ddev 
 Container ddev-ssh-agent  Recreate 
 Container ddev-ssh-agent  Recreated 
 Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
Using custom nginx configuration in /Users/dmckenna/Sites/myproject/.ddev/nginx_full/nginx-site.conf 
Using nginx snippets: [/Users/dmckenna/Sites/myproject/.ddev/nginx/file-redirect.conf /Users/dmckenna/Sites/myproject/.ddev/nginx/patternlab.conf] 
Using custom mysql configuration: [/Users/dmckenna/Sites/myproject/.ddev/mysql/migration.cnf] 
Custom configuration is updated on restart.
If you don't see your custom configuration taking effect, run 'ddev restart'. 
 Container ddev-myproject-redis  Created 
 Container ddev-myproject-db  Recreate 
 Container ddev-myproject-solr  Created 
 Container ddev-myproject-web  Recreate 
 Container ddev-myproject-db  Recreated 
 Container ddev-myproject-web  Recreated 
 Container ddev-myproject-db  Started 
 Container ddev-myproject-redis  Started 
 Container ddev-myproject-solr  Started 
 Container ddev-myproject-web  Started 
Failed waiting for web/db containers to become ready: web container failed: log=, err=health check timed out after 2m0s: labels map[com.ddev.site-name:myproject com.docker.compose.service:web] timed out without becoming healthy, status= 
 Container ddev-router  Created 
 Container ddev-router  Started 
Failed to start myproject: container(s) failed to become healthy before their configured timeout or in 120 seconds. This may be just a problem with the healthcheck and not a functional problem. (health check timed out: labels map[com.ddev.site-name:myproject] timed out without becoming healthy, status=, detail= ddev-myproject-web:starting - more info with `docker inspect --format "{{json .State.Health }}" ddev-myproject-web` ) 
```

**Comment by damienmckenna** [comment]
FWIW when I start Colima it has this error:

```
{"level":"info","msg":"Waiting for the essential requirement 1 of 3: \"ssh\"","time":"2023-08-25T10:56:59-04:00"}
{"level":"debug","msg":"executing script \"ssh\"","time":"2023-08-25T10:56:59-04:00"}
{"level":"info","msg":"[VZ] - vm state change: running","time":"2023-08-25T10:56:59-04:00"}
{"level":"debug","msg":"executing ssh for script \"ssh\": /usr/bin/ssh [ssh -F /dev/null -o IdentityFile=\"/Users/dmckenna/.lima/_config/user\" -o IdentityFile=\"/Users/dmckenna/.ssh/id_rsa\" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o NoHostAuthenticationForLocalhost=yes -o GSSAPIAuthentication=no -o PreferredAuthentications=publickey -o Compression=no -o BatchMode=yes -o IdentitiesOnly=yes -o Ciphers=\"^aes128-gcm@openssh.com,aes256-gcm@openssh.com\" -o User=dmckenna -o ControlMaster=auto -o ControlPath=\"/Users/dmckenna/.lima/colima/ssh.sock\" -o ControlPersist=5m -p 49544 127.0.0.1 -- /bin/bash]","time":"2023-08-25T10:56:59-04:00"}
{"level":"debug","msg":"stdout=\"\", stderr=\"ssh: connect to host 127.0.0.1 port 49544: Connection refused\\r\\n\", err=failed to execute script \"ssh\": stdout=\"\", stderr=\"ssh: connect to host 127.0.0.1 port 49544: Connection refused\\r\\n\": exit status 255","time":"2023-08-25T10:56:59-04:00"}
{"level":"info","msg":"Waiting for the essential requirement 1 of 3: \"ssh\"","time":"2023-08-25T10:57:09-04:00"}
{"level":"debug","msg":"executing script \"ssh\"","time":"2023-08-25T10:57:09-04:00"}
{"level":"debug","msg":"executing ssh for script \"ssh\": /usr/bin/ssh [ssh -F /dev/null -o IdentityFile=\"/Users/dmckenna/.lima/_config/user\" -o IdentityFile=\"/Users/dmckenna/.ssh/id_rsa\" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o NoHostAuthenticationForLocalhost=yes -o GSSAPIAuthentication=no -o PreferredAuthentications=publickey -o Compression=no -o BatchMode=yes -o IdentitiesOnly=yes -o Ciphers=\"^aes128-gcm@openssh.com,aes256-gcm@openssh.com\" -o User=dmckenna -o ControlMaster=auto -o ControlPath=\"/Users/dmckenna/.lima/colima/ssh.sock\" -o ControlPersist=5m -p 49544 127.0.0.1 -- /bin/bash]","time":"2023-08-25T10:57:09-04:00"}
2023/08/25 10:57:10 tcpproxy: for incoming conn 127.0.0.1:49546, error dialing "192.168.5.15:22": connect tcp 192.168.5.15:22: connection was refused
{"level":"debug","msg":"stdout=\"\", stderr=\"kex_exchange_identification: read: Connection reset by peer\\r\\nConnection reset by 127.0.0.1 port 49544\\r\\n\", err=failed to execute script \"ssh\": stdout=\"\", stderr=\"kex_exchange_identification: read: Connection reset by peer\\r\\nConnection reset by 127.0.0.1 port 49544\\r\\n\": exit status 255","time":"2023-08-25T10:57:10-04:00"}
```

**Comment by damienmckenna** [comment]
Related: [lima#1200](https://github.com/lima-vm/lima/issues/1200)

**Comment by damienmckenna** [comment]
Clarified the issue title that this affects both Mutagen and nfs.

**Comment by damienmckenna** [comment]
Here's the output from `ddev debug test` after switching to nfs:

```
ddev debug test
Running bash [-c /var/folders/0m/tg6xfqy14mn4lnws2jmstr4w0000gr/T/test_ddev.sh] 
======= Existing project config =========
These config files were loaded for project myproject: [/Users/dmckenna/Sites/myproject/.ddev/config.yaml]
name: myproject
type: drupal9
docroot: web
php_version: 8.0
webserver_type: nginx-fpm
webimage: ddev/ddev-webserver:20230803_php_serialize_precision
router_http_port: 80
router_https_port: 443
additional_hostnames: []
additional_fqdns: [local.pe.gatech.edu]
mariadb_version: 10.3
database: {mariadb 10.3}
performance_mode: nfs
hooks: map[post-import-db:[map[exec:drush cr] map[exec:drush en -y stage_file_proxy] map[exec:drush sql-sanitize -y] map[exec:drush uli]] post-pull:[map[exec:drush cr] map[exec:drush en -y stage_file_proxy] map[exec:drush sql-sanitize -y] map[exec:drush uli]] post-start:[map[exec:mysql -uroot -proot -e "CREATE DATABASE IF NOT EXISTS d7; GRANT ALL ON d7.* to 'db'@'%';" service:db]]]
mailhog_port: 8025
mailhog_https_port: 8026
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2
nodejs_version: 18
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-475 in ../tryddevproject-475 =========


Please run 'ddev debug test' in the root of the existing project where you're having trouble.

Failed running test_ddev.sh: exit status 4
. You can run it manually with `curl -sL -O https://raw.githubusercontent.com/ddev/ddev/master/cmd/ddev/cmd/scripts/test_ddev.sh && bash test_ddev.sh` 
```

**Comment by rfay** [comment]
Hi @damienmckenna  - You probably know already that Colima/Lima has had several bumps in the road lately, see 
* https://github.com/abiosoft/colima/issues/786#issuecomment-1693630416
* https://github.com/Homebrew/homebrew-core/issues/140244
* https://github.com/Homebrew/brew/pull/15903

You might want to use the workarounds or try [Orbstack](https://orbstack.dev) in the meantime. 

When Colima is working, DDEV is working fine.

**Comment by damienmckenna** [comment]
Orbstack worked flawlessly for one project, thank you for suggesting it, I wasn't aware of that one.

Now to test it out on a larger project..

**Comment by rfay** [comment]
Closing this, because it has theoretically be fixed upstream.

Where it's it's not fixed, `brew reinstall --force qemu` seems to fix it.
